homeassistant/components/ecoforest/coordinator.py

```diff
@@ -0,0 +1,49 @@
+"""The ecoforest coordinator."""
+
+
+import logging
+
+import async_timeout
+from pyecoforest.api import EcoforestApi
+from pyecoforest.exceptions import EcoforestAuthenticationRequired, EcoforestError
+from pyecoforest.models.device import Device
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import POLLING_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EcoforestCoordinator(DataUpdateCoordinator[Device]):
+    """DataUpdateCoordinator to gather data from ecoforest device."""
+
+    def __init__(self, hass: HomeAssistant, api: EcoforestApi, host: str) -> None:
+        """Initialize DataUpdateCoordinator."""
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="ecoforest",
+            update_interval=POLLING_INTERVAL,
+        )
+        self.api = api
+        self.host = host
```

Comment:
You're not using host anywhere, you can omit it

Reply:
I use the coordinator.host as way to pass device host information, in this particular case it's used by the entity to set the device identifier. Maybe we could find a better way but I felt this way this info would be shared using the coordinator only.

Reply:
Do you believe this is blocker?

Reply:
I would prefer to actually use the config entry unique id in this case, which will be the device serial. That's always unique :) Even when someone changes routers or uses DHCP.

Reply:
I also thought about it, I was about to change it to use the serial number. 👍